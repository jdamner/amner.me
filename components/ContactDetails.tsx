"use client";

import React from "react";

/* API */
import { useState } from "react";

/* Components */
import { EnvelopeOpenIcon, EnvelopeIcon } from "@heroicons/react/24/outline";
import Title from "./Atoms/Title";
import ProtectedLink from "./Atoms/Links/ProtectedLink";

export default function ContactDetails(
  props: React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLDivElement>,
    HTMLDivElement
  >,
) {
  const [isMouseOver, setIsMouseOver] = useState(false);
  const enableMouseOver = () => setIsMouseOver(true);
  const disableMouseOver = () => setIsMouseOver(false);

  const iconClass = "inline-block w-12 h-12 mr-3";

  return (
    <section className="px-4 py-3 container mx-auto" id="contact" {...props}>
      <Title title="Contact Me">Say hello</Title>
      <div className="w-full gap-5 my-10 text-center">
        <p className="my-5">
          If you&apos;d like to get in touch, just send me an email by clicking
          the link below.
        </p>
        <ProtectedLink
          href="mailto:jdamner@me.com"
          className="block bg-white dark:bg-slate-800 border-2 px-3 py-2 rounded-sm font-bold border-solid border-slate-900 hover:bg-slate-100 dark:hover:bg-slate-700"
          onMouseEnter={enableMouseOver}
          onMouseLeave={disableMouseOver}
        >
          {isMouseOver ? (
            <EnvelopeOpenIcon className={iconClass} />
          ) : (
            <EnvelopeIcon className={iconClass} />
          )}
          Send an email
        </ProtectedLink>
      </div>
    </section>
  );
}
