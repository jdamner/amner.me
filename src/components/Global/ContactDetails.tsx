
import Title from "./Title";
import { EnvelopeOpenIcon, EnvelopeIcon } from "@heroicons/react/24/outline";
import Link from "next/link";
import { useState } from "react";

export default function ContactDetails(): JSX.Element {

	const [isMouseOver, setIsMouseOver] = useState(false);

	const enableMouseOver = () => {
		setIsMouseOver(true);
	}

	const disableMouseOver = () => {
		setIsMouseOver(false);
	}

	return (
		<section className='bg-white px-4 py-3 container mx-auto' id='contact'>
			<Title title="Contact Me">Say hello</Title>
			<div className='w-full gap-5 my-10 text-center'>
				<p className='my-5'>If you&apos;d like to get in touch, just send me an email by clicking the link below.</p>
				<Link href='mailto:jdamner@me.com' className="block bg-white border-2 px-3 py-2 rounded-sm font-bold border-solid border-slate-900 hover:bg-slate-100" onMouseEnter={enableMouseOver} onMouseLeave={disableMouseOver}>
					{isMouseOver ? <EnvelopeOpenIcon className='inline-block w-12 h-14' /> : <EnvelopeIcon className='inline-block w-12 h-14' />}
					Send an email
				</Link>
			</div>
		</section>
	)
}
