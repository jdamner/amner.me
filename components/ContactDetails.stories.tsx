import type { Meta, StoryObj } from "@storybook/react";

import ContactDetails from "./ContactDetails";

type Story = StoryObj<typeof ContactDetails>;

export default {
  title: "Components/ContactDetails",
  component: ContactDetails,
} as Meta;

export const Default: Story = {};
