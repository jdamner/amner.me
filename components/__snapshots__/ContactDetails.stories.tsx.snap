// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/ContactDetails Default smoke-test 1`] = `
<div class="bg-black text-orange-100  font-sans">
  <section class="col-span-2 mt-5 lg:ml-8"
           id="contact"
  >
    <div class="px-4 py-3 bg-highlighted text-black flex flex-col gap-5">
      <h2 class="text-9xl tracking-tight uppercase"
          style="font-family: &quot;Bebas Neue&quot;; font-style: normal; font-weight: 400;"
      >
        Say hello
      </h2>
      <p class="mb-5">
        If you'd like to get in touch, just send me an email by clicking the link below.
      </p>
      <a class="inline-block px-3 py-3 bg-orange-100 text-black border-4 border-orange-600  hover:bg-orange-600 uppercase font-black text-4xl"
         href="/#"
         style="font-family: &quot;Bebas Neue&quot;; font-style: normal; font-weight: 400;"
      >
        <svg xmlns="http://www.w3.org/2000/svg"
             fill="none"
             viewbox="0 0 24 24"
             stroke-width="1.5"
             stroke="currentColor"
             aria-hidden="true"
             data-slot="icon"
             class="inline-block w-12 h-12 mr-3"
        >
          <path stroke-linecap="round"
                stroke-linejoin="round"
                d="M21.75 6.75v10.5a2.25 2.25 0 0 1-2.25 2.25h-15a2.25 2.25 0 0 1-2.25-2.25V6.75m19.5 0A2.25 2.25 0 0 0 19.5 4.5h-15a2.25 2.25 0 0 0-2.25 2.25m19.5 0v.243a2.25 2.25 0 0 1-1.07 1.916l-7.5 4.615a2.25 2.25 0 0 1-2.36 0L3.32 8.91a2.25 2.25 0 0 1-1.07-1.916V6.75"
          >
          </path>
        </svg>
        Send an email
      </a>
    </div>
  </section>
</div>
`;
